Jenkinsfile

```diff
@@ -1017,9 +1017,75 @@ pipeline {
                                                       id: 'VM_test')
                         }
                     }
-                }
-            }
-        }
+                } // End run_test.sh stage
+                stage('run_test_memcheck.sh') {
+                    when {
+                      beforeAgent true
+                      expression { ! skip_stage('run_test') }
+                    }
+                    agent {
+                        label 'ci_vm1'
+                    }
+                    steps {
+                        script {
+                            if (quickbuild()) {
+                                // TODO: these should be gotten from the Requires: of RPMs
+                                qb_inst_rpms = " spdk-tools mercury boost-devel"
+                            }
+                        }
+                        provisionNodes NODELIST: env.NODELIST,
+                                       node_count: 1,
+                                       profile: 'daos_ci',
+                                       distro: 'el7',
+                                       snapshot: true,
+                                       inst_repos: el7_component_repos + ' ' +
+                                                   component_repos(),
+                                       inst_rpms: 'gotestsum openmpi3 ' +
+                                                  'hwloc-devel argobots ' +
+                                                  'fuse3-libs fuse3 ' +
+                                                  'libisa-l-devel libpmem ' +
+                                                  'libpmemobj protobuf-c ' +
+                                                  'spdk-devel libfabric-devel '+
+                                                  'pmix numactl-devel ' +
+                                                  'libipmctl-devel' +
+                                                  qb_inst_rpms
+                        timeout(time:60, unit:'MINUTES') {
+                          runTest stashes: [ 'centos7-gcc-tests',
+                                           'centos7-gcc-install',
+                                           'centos7-gcc-build-vars' ],
+                                script: "SSH_KEY_ARGS=${env.SSH_KEY_ARGS} " +
+                                        "NODELIST=${env.NODELIST} " +
+                                        'ci/run_test_memcheck.sh'
```

Comment:
Given the duplication between this new stage and the existing one, these duplicated blocks need to be refactored into common code using [pipeline-lib](https://github.com/daos-stack/pipeline-lib) similar to the refactor that is being done in #2743.  This is required as the size that this file is allowed to grow to is limited and as such we cannot allow duplication in it.